File: 345. Reverse Vowels Of A String/main.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = {'a','e','i','o','u','A','E','I','O','U'}
        s_list = list(s)
        i, j = 0, len(s) - 1

        while i < j:
            if s_list[i] in vowels:
                while s_list[j] not in vowels and i < j:
                    j -= 1
                s_list[i], s_list[j] = s_list[j], s_list[i]
                j -= 1
            i += 1
        return ''.join(s_list)

File: 345. Reverse Vowels Of A String/main.cpp
#include <string>
using namespace std;

class Solution {
public:
    string reverseVowels(string s) {
        int index= 0;
        for(int i=0; i<s.length();i++){
            if (s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u'||s[i]=='A'||s[i]=='E'||s[i]=='I'||s[i]=='O'||s[i]=='U'){
                for(int j=(s.length()-index);j>=i;j--){
                    index+=1;
                    if(s[j]=='a'||s[j]=='e'||s[j]=='i'||s[j]=='o'||s[j]=='u'||s[j]=='A'||s[j]=='E'||s[j]=='I'||s[j]=='O'||s[j]=='U'){
                        char temp = s[i];
                        s[i]=s[j];
                        s[j]=temp;
                        break;
                    }
                }     
            }
        }
        return s;
    }
};